### PR TITLE
rework maintenance script to be able to patch golang binaries

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Allow to patch Golang binaries through the maintenance branch

# Impacts
Maintenance script

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Can now patch the Golang binaries through the maintenance script (addons/pf-maint.pl)

### CODE REVIEW

File: addons/pf-maint.pl [hunk 1]
```diff
@@ -67,6 +67,12 @@ =head1 SYNOPSIS
     "t/*",
 );
 
+our @patchable_binaries = (
+    "pfhttpd",
+    "pfdns",
+    "pfdhcp",
```
Comment (extrafu) [inline addons/pf-maint.pl:73]: How about pfstats?

File: addons/pf-maint.pl [hunk 2]
```diff
@@ -224,6 +244,25 @@ sub print_dot {
...
+        exit;
+    }
+}
+
+sub download_and_install_binaries {
+    foreach my $binary (@patchable_binaries) {
+        print "Performing patching of $binary.......\n";
+        my $data = get_url("$BASE_BINARIES_URL/maintenance/$PF_RELEASE/$binary");
+        rename("$BINARIES_DIRECTORY/$binary", "$BINARIES_DIRECTORY/$binary-pre-maintenance") or die "Cannot backup binary: $!\n";
+        write_file("/usr/local/pf/bin/$binary", $data);
```
Comment (extrafu) [inline addons/pf-maint.pl]: Proper error-handling there too?

File: addons/pf-maint.pl (hunk 1 above)
Comment (julsemaan) [reply addons/pf-maint.pl:73]: Just added it

File: addons/pf-maint.pl [hunk 3]
```diff
@@ -268,6 +268,7 @@ sub download_and_install_binaries {
 
         rename($binary_path, "$binary_path-pre-maintenance") or die "Cannot backup binary: $!\n";
         rename("$binary_path-maintenance-decrypted", $binary_path) or die "Cannot install binary: $!\n";
+        unlink("$binary_path-maintenance-encrypted") or warn "Couldn't delete temporary download file, everything will keep working but the stale file will still be there ($!)\n";
         chmod 0755, "$binary_path";
```
Comment (jrouzierinverse) [inline addons/pf-maint.pl]: Change the ownership of the files to match what ever it is when it is packaged.

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by julsemaan** [comment]
@jrouzierinverse and @fdurand 

Please perform a final review and approve the PR if everything is fine.

I'll perform the merge since I'll be testing it right after putting it in the maintenance

File: addons/pf-maint.pl [hunk 4]
```diff
@@ -256,6 +259,14 @@ sub accept_binary_patching {
     }
 }
 
+sub install_binary_sign_key_if_needed {
+    my $rc = system("gpg --list-keys | grep $BINARIES_SIGN_KEY_ID");
```
Comment (jrouzierinverse) [inline addons/pf-maint.pl]: use 
```bash 
gpg --list-keys $BINARIES_SIGN_KEY_ID
```
To avoid additional call to grep

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
Tested on debian, ok to merge

**Comment by julsemaan** [comment]
@jrouzierinverse 

Adjusted based on the comments

**Review by jrouzierinverse — APPROVED** [review]

**Pull request merged**